Source/Urho3D/Input/FreeFlyController.cpp

```diff
@@ -31,17 +31,15 @@
 #include "../Input/InputEvents.h"
 #include "../Scene/Node.h"
 #include "../UI/UI.h"
+#include "../Engine/StateManager.h"
 
 namespace Urho3D
 {
 namespace
 {
-
-float ApplyDeadZone(float value, float deadZone)
+float ClampPitch(float pitch)
 {
-    if (value >= -deadZone && value <= deadZone)
-        return 0.0f;
-    return (value - Sign(value) * deadZone) / (1.0f - deadZone);
+    return Clamp(pitch, -90.0f, 90.0f);
```

Comment:
There may be a problem with high pitches. Let's make this constant configurable.

Reply:
I've fixed gimbal lock with one of previous PRs but if you still concerned - sure.